### PR TITLE
Docs, Tests, Coverage

### PR DESCRIPTION
**Pull request by Edouard360**
* Docs and tests added.
* Build of halite in Travis enabled (compiling option added for g++-4.9)
* Coverage
* Invariance of the Bot class

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,21 +5,31 @@ language: python
...
+  - pip install -r requirements.txt
+  - make
 
 script:
   # Tests
   - python -m unittest discover -v
   # Style checks
   - pylint train/experience.py
+  # Coverage checks
+  - py.test --cov=train tests/
```
Comment (lremus) [inline .travis.yml:29]: @Edouard360 
`py.test` is only for coverage tests and not unit/continuous/regression tests am I right ?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
-[![Build Status](https://travis-ci.org/louis-r/Halite-Python-RL.svg?branch=master)](https://travis-ci.org/louis-r/Halite-Python-RL)
+[![Build Status](https://travis-ci.org/louis-r/Halite-Python-RL.svg?branch=master)](https://travis-ci.org/louis-r/Halite-Python-RL) [![Coverage Status](https://coveralls.io/repos/github/Edouard360/Halite-Python-RL/badge.svg?branch=master)](https://coveralls.io/github/Edouard360/Halite-Python-RL?branch=master) [![Documentation Status](https://readthedocs.org/projects/halite-python-rl/badge/?version=latest)](http://halite-python-rl.readthedocs.io/en/latest/?badge=latest)
```
Comment (lremus) [inline README.md:1]: @Edouard360 
Be careful you still have the build status of my fork here.

**Comment by lremus** [comment]
Very nice PR with clear commit messages :)

I see that you edited python files, are they pylint-compliant ?
If so, make sure to add them to .travis_yaml pylint checks 👍

File: README.md (hunk 2 above)
Comment (Edouard360) [reply README.md:1]: Indeed ! I'll correct it 👍

**Comment by Edouard360** [comment]
Great idea 😄 
Not checked any of them yet

File: .travis.yml (hunk 1 above)
Comment (Edouard360) [reply .travis.yml:29]: I think unit tests are included. The tests in tests/reward_test.py are unit and executed through this...

**Pull request merged**